### ISSUE TITLE
The training loss does not converge

### ISSUE THREAD

**Issue by ZhengtianXu** [open-mmlab/mmhuman3d#79]
I have trained models like SPIN, but the training loss (especially keypoint_2dloss) is not converge, which is getting bigger. I just download COCO dataset and modify the dataset setting in the config file. 

```
2022-01-17 14:43:00,158 - mmhuman3d - INFO - Epoch [1][50/3125] lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 22:28:06, time: 2.593, data_time: 0.067, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 0.2118, vertex_loss: 0.2076, smpl_pose_loss: 0.1244, smpl_betas_loss: 0.0138, camera_loss: 0.5341, loss: 1.0918
2022-01-17 14:45:06,227 - mmhuman3d - INFO - Epoch [1][100/3125]        lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 22:07:29, time: 2.521, data_time: 0.010, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 0.6550, vertex_loss: 0.2113, smpl_pose_loss: 0.1248, smpl_betas_loss: 0.0168, camera_loss: 0.0009, loss: 1.0087
2022-01-17 14:47:11,979 - mmhuman3d - INFO - Epoch [1][150/3125]        lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:58:06, time: 2.515, data_time: 0.010, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 0.9062, vertex_loss: 0.2161, smpl_pose_loss: 0.1248, smpl_betas_loss: 0.0206, camera_loss: 0.0639, loss: 1.3317
2022-01-17 14:49:19,096 - mmhuman3d - INFO - Epoch [1][200/3125]        lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:55:54, time: 2.542, data_time: 0.010, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 1.0589, vertex_loss: 0.2169, smpl_pose_loss: 0.1217, smpl_betas_loss: 0.0301, camera_loss: 0.0000, loss: 1.4275
2022-01-17 14:51:24,083 - mmhuman3d - INFO - Epoch [1][250/3125]        lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:49:20, time: 2.500, data_time: 0.010, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 1.5051, vertex_loss: 0.2175, smpl_pose_loss: 0.1195, smpl_betas_loss: 0.0256, camera_loss: 0.0307, loss: 1.8983
2022-01-17 14:53:30,917 - mmhuman3d - INFO - Epoch [1][300/3125]        lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:47:26, time: 2.537, data_time: 0.011, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 1.4352, vertex_loss: 0.2168, smpl_pose_loss: 0.1235, smpl_betas_loss: 0.0221, camera_loss: 0.0000, loss: 1.7976
2022-01-17 14:55:38,974 - mmhuman3d - INFO - Epoch [1][350/3125]        lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:47:16, time: 2.561, data_time: 0.010, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 1.9522, vertex_loss: 0.2104, smpl_pose_loss: 0.1221, smpl_betas_loss: 0.0106, camera_loss: 0.0000, loss: 2.2953
2022-01-17 14:57:46,203 - mmhuman3d - INFO - Epoch [1][400/3125]        lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:45:33, time: 2.545, data_time: 0.011, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 2.6034, vertex_loss: 0.2096, smpl_pose_loss: 0.1228, smpl_betas_loss: 0.0099, camera_loss: 0.0000, loss: 2.9457
2022-01-17 14:59:56,095 - mmhuman3d - INFO - Epoch [1][450/3125]        lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:46:47, time: 2.598, data_time: 0.010, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 2.5713, vertex_loss: 0.2144, smpl_pose_loss: 0.1229, smpl_betas_loss: 0.0227, camera_loss: 0.0000, loss: 2.9314
2022-01-17 15:02:02,843 - mmhuman3d - INFO - Epoch [1][500/3125]        lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:44:07, time: 2.535, data_time: 0.010, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 3.3346, vertex_loss: 0.2117, smpl_pose_loss: 0.1235, smpl_betas_loss: 0.0110, camera_loss: 0.0001, loss: 3.6810
2022-01-17 15:04:10,546 - mmhuman3d - INFO - Epoch [1][550/3125]        lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:42:26, time: 2.554, data_time: 0.011, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 3.5986, vertex_loss: 0.2122, smpl_pose_loss: 0.1161, smpl_betas_loss: 0.0237, camera_loss: 0.0003, loss: 3.9508
2022-01-17 15:06:18,414 - mmhuman3d - INFO - Epoch [1][600/3125]        lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:40:49, time: 2.557, data_time: 0.011, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 4.4794, vertex_loss: 0.2034, smpl_pose_loss: 0.1147, smpl_betas_loss: 0.0097, camera_loss: 0.0001, loss: 4.8073
2022-01-17 15:08:27,190 - mmhuman3d - INFO - Epoch [1][650/3125]        lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:39:50, time: 2.575, data_time: 0.011, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 3.7132, vertex_loss: 0.2025, smpl_pose_loss: 0.1140, smpl_betas_loss: 0.0093, camera_loss: 0.0000, loss: 4.0391
2022-01-17 15:10:38,125 - mmhuman3d - INFO - Epoch [1][700/3125]        lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:40:15, time: 2.619, data_time: 0.012, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 4.4091, vertex_loss: 0.2049, smpl_pose_loss: 0.1173, smpl_betas_loss: 0.0108, camera_loss: 0.0000, loss: 4.7421
2022-01-17 15:12:44,682 - mmhuman3d - INFO - Epoch [1][750/3125]        lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:37:21, time: 2.531, data_time: 0.011, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 4.4774, vertex_loss: 0.2119, smpl_pose_loss: 0.1186, smpl_betas_loss: 0.0125, camera_loss: 0.0000, loss: 4.8204
2022-01-17 15:14:46,391 - mmhuman3d - INFO - Epoch [1][800/3125]        lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:31:29, time: 2.434, data_time: 0.010, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 5.5440, vertex_loss: 0.2023, smpl_pose_loss: 0.1113, smpl_betas_loss: 0.0096, camera_loss: 0.0002, loss: 5.8674
2022-01-17 15:16:54,684 - mmhuman3d - INFO - Epoch [1][850/3125]        lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:30:00, time: 2.566, data_time: 0.012, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 6.3298, vertex_loss: 0.2075, smpl_pose_loss: 0.1155, smpl_betas_loss: 0.0117, camera_loss: 0.0001, loss: 6.6646
2022-01-17 15:19:03,403 - mmhuman3d - INFO - Epoch [1][900/3125]        lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:28:40, time: 2.574, data_time: 0.011, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 8.4531, vertex_loss: 0.2038, smpl_pose_loss: 0.1110, smpl_betas_loss: 0.0181, camera_loss: 0.0001, loss: 8.7860
2022-01-17 15:21:09,466 - mmhuman3d - INFO - Epoch [1][950/3125]        lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:25:51, time: 2.521, data_time: 0.011, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 6.5281, vertex_loss: 0.2075, smpl_pose_loss: 0.1143, smpl_betas_loss: 0.0211, camera_loss: 70.7021, loss: 77.5730
2022-01-17 15:23:19,187 - mmhuman3d - INFO - Epoch [1][1000/3125]       lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:24:56, time: 2.594, data_time: 0.012, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 7.2353, vertex_loss: 0.2082, smpl_pose_loss: 0.1164, smpl_betas_loss: 0.0147, camera_loss: 0.0000, loss: 7.5747
2022-01-17 15:25:30,366 - mmhuman3d - INFO - Epoch [1][1050/3125]       lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:24:36, time: 2.624, data_time: 0.011, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 7.4498, vertex_loss: 0.2072, smpl_pose_loss: 0.1169, smpl_betas_loss: 0.0154, camera_loss: 0.0000, loss: 7.7893
2022-01-17 15:27:37,373 - mmhuman3d - INFO - Epoch [1][1100/3125]       lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:22:12, time: 2.540, data_time: 0.010, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 7.3770, vertex_loss: 0.2133, smpl_pose_loss: 0.1166, smpl_betas_loss: 0.0297, camera_loss: 0.0000, loss: 7.7366
2022-01-17 15:29:49,020 - mmhuman3d - INFO - Epoch [1][1150/3125]       lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:21:51, time: 2.633, data_time: 0.011, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 8.5110, vertex_loss: 0.2076, smpl_pose_loss: 0.1148, smpl_betas_loss: 0.0171, camera_loss: 0.0000, loss: 8.8505
2022-01-17 15:31:57,100 - mmhuman3d - INFO - Epoch [1][1200/3125]       lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:19:51, time: 2.562, data_time: 0.011, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 9.0734, vertex_loss: 0.2063, smpl_pose_loss: 0.1150, smpl_betas_loss: 0.0197, camera_loss: 0.0000, loss: 9.4143
2022-01-17 15:34:06,728 - mmhuman3d - INFO - Epoch [1][1250/3125]       lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:18:28, time: 2.593, data_time: 0.012, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 10.2803, vertex_loss: 0.2078, smpl_pose_loss: 0.1174, smpl_betas_loss: 0.0135, camera_loss: 0.0000, loss: 10.6190
2022-01-17 15:36:16,439 - mmhuman3d - INFO - Epoch [1][1300/3125]       lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:17:03, time: 2.594, data_time: 0.011, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 13.1727, vertex_loss: 0.2084, smpl_pose_loss: 0.1168, smpl_betas_loss: 0.0157, camera_loss: 0.0000, loss: 13.5136
2022-01-17 15:38:28,994 - mmhuman3d - INFO - Epoch [1][1350/3125]       lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:16:38, time: 2.651, data_time: 0.011, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 10.8980, vertex_loss: 0.2077, smpl_pose_loss: 0.1145, smpl_betas_loss: 0.0139, camera_loss: 0.0000, loss: 11.2341
2022-01-17 15:40:33,621 - mmhuman3d - INFO - Epoch [1][1400/3125]       lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:13:16, time: 2.493, data_time: 0.011, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 11.8481, vertex_loss: 0.2136, smpl_pose_loss: 0.1143, smpl_betas_loss: 0.0194, camera_loss: 0.0000, loss: 12.1954
2022-01-17 15:42:34,076 - mmhuman3d - INFO - Epoch [1][1450/3125]       lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:08:34, time: 2.409, data_time: 0.011, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 10.6372, vertex_loss: 0.2035, smpl_pose_loss: 0.1077, smpl_betas_loss: 0.0151, camera_loss: 0.0000, loss: 10.9635
2022-01-17 15:44:44,637 - mmhuman3d - INFO - Epoch [1][1500/3125]       lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:07:22, time: 2.611, data_time: 0.010, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 11.9690, vertex_loss: 0.2182, smpl_pose_loss: 0.1221, smpl_betas_loss: 0.0177, camera_loss: 0.0000, loss: 12.3270
2022-01-17 15:46:54,947 - mmhuman3d - INFO - Epoch [1][1550/3125]       lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:06:03, time: 2.606, data_time: 0.010, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 11.0126, vertex_loss: 0.2086, smpl_pose_loss: 0.1100, smpl_betas_loss: 0.0139, camera_loss: 0.0000, loss: 11.3451
2022-01-17 15:49:08,060 - mmhuman3d - INFO - Epoch [1][1600/3125]       lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:05:32, time: 2.662, data_time: 0.011, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 15.5035, vertex_loss: 0.2117, smpl_pose_loss: 0.1197, smpl_betas_loss: 0.0119, camera_loss: 0.0000, loss: 15.8468
2022-01-17 15:51:18,177 - mmhuman3d - INFO - Epoch [1][1650/3125]       lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:04:01, time: 2.602, data_time: 0.010, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 15.4472, vertex_loss: 0.2105, smpl_pose_loss: 0.1179, smpl_betas_loss: 0.0120, camera_loss: 0.0000, loss: 15.7876
2022-01-17 15:53:28,477 - mmhuman3d - INFO - Epoch [1][1700/3125]       lr_backbone: 3.000e-05 lr_head: 3.000e-05, eta: 21:02:31, time: 2.606, data_time: 0.010, memory: 3186, keypoints3d_loss: 0.0000, keypoints2d_loss: 16.4281, vertex_loss: 0.2120, smpl_pose_loss: 0.1165, smpl_betas_loss: 0.0136, camera_loss: 0.0000, loss: 16.7702
```

**Comment by caizhongang** [comment]
Hi @ZhengtianXu , did you use a pretrained model (`spin_pretrain.pth`) in the training?

**Comment by ZhengtianXu** [comment]
> Hi @ZhengtianXu , did you use a pretrained model (`spin_pretrain.pth`) in the training?

@caizhongang Yes, I use the pretrained model in the training. In the config file, I just comment other datasets and use coco and lsp datasets. The partition is [0.3, 0.7].

**Comment by caizhongang** [comment]
COCO and LSP do not provide the SMPL annotations that are quite important to training of SPIN. I would suggest follow the config and use H36M with mosh (unfortunately we are not allowed to provide the mosh data due to license issue at the current moment, but there may be other sources).

Meanwhile we have also received reports on the 2D keypoint loss divergence issue, the team is looking into it.
